### PR TITLE
Log a warning for a duplicate sid

### PR DESCRIPTION
**Pull request by megha070**
Make sure these boxes are signed before submitting your Pull Request
-- thank you.

- [* ] I have read the contributing guide lines at
  https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [ *] I have signed the Open Information Security Foundation
  contribution agreement at
  https://suricata-ids.org/about/contribution-agreement/
- [ *] I have updated the user guide (in doc/userguide/) to reflect the
  changes made (if applicable)

Link
to
[redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:

Describe changes:
Made changes to build_rule_map function of main.py file 
such that whenever a duplicate sid is encountered a warning is logged.

### CODE REVIEW

File: suricata/update/main.py [hunk 1]
```diff
@@ -687,13 +687,23 @@ def build_rule_map(rules):
...
     rulemap = {}
-
+    
     for rule in rules:
         if rule.id not in rulemap:
             rulemap[rule.id] = rule
+            FORMAT = '%(version)s %(message)s'
+            logging.basicConfig(format=FORMAT)
+            d={'version: rule'}
+            logger = logging.getLogger('tcpserver')
```
Comment (shivan1b) [inline suricata/update/main.py]: You do not need to configure logger multiple times, just loop up how it has been used in the rest of the code. Now. could you please write down the steps you're using for testing? Make sure to write the steos down after having tried them on your system.

Comment (shivan1b) [reply suricata/update/main.py]: In addition to this, I would also request you to have a look at the Travis logs and understand what have you done incorrectly (logically).

**Comment by shivan1b** [comment]
Hi @megha070 !
Are you planning on implementing the changes requested and submitting a new PR? Please let us know. Thanks for your contribution. :)

**Comment by megha070** [comment]
I would love to but I am still facing problems with setting up the environment, hence couldn't test my code.

**Comment by vagishagupta23** [comment]
> I would love to but I am still facing problems with setting up the environment, hence couldn't test my code.
Hi @megha070. Thanks for letting us know. I'll be helping you set the environment. Could you please tell if you have followed all steps mentioned in ([link](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Ubuntu_Installation_from_GIT))? If yes, did you try running suricata-update with `sudo`? Could you please share the output you get running `sudo ./bin/suricata-update`?

**Comment by shivan1b** [comment]
Replaced with #160

**Pull request closed without merging**